llvm/lib/Transforms/Utils/InlineFunction.cpp

```diff
@@ -1398,7 +1398,7 @@ static void AddParamAndFnBasicAttributes(const CallBase &CB,
         if (!Arg)
           continue;
 
-        if (AL.hasParamAttr(I, Attribute::ByVal))
+        if (NewInnerCB->getParamAttr(I, Attribute::ByVal).isValid())
```

Comment:
why not use the paramHasAttr funktion instead of creating a Attribute and then check if it is valid?
```suggestion
        if (NewInnerCB->paramHasAttr(I, Attribute::ByVal))
```